### PR TITLE
chore(flask): avoid using deprecated internals

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

The following flask attributes have been deprecated in flask v2.2.2 and will be removed in the next minor release:
- [flask. _app_ctx_stack](https://github.com/pallets/flask/blob/2a54cfa5cecf4dd9f17525bf3ebae4d33a155776/src/flask/__init__.py#L58) 
- [flask. _request_ctx_stack](https://github.com/pallets/flask/blob/2a54cfa5cecf4dd9f17525bf3ebae4d33a155776/src/flask/__init__.py#L69)

This change replaces deprecated methods with flask globals (flask.current_app and flask.request). 

Context: https://github.com/pallets/flask/pull/4682

## Checklist
- [x] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Relevant issue(s)
Closes: https://github.com/DataDog/dd-trace-py/issues/4430

## Testing strategy
Ensure existing flask tests pass

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by Kyle-Verhoog — APPROVED** [review]
happy to see `LocalStack` go 

lgtm given that this works for older flask versions as well (which our tests should verify)

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]
we should have a release note here in case someone on an older version has issues. If so then when searching our release notes they'll see that the issue has been addressed in a newer version.

File: releasenotes/notes/remove-deprecated-flask-methods-8d7a4a3d32c3c69d.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    flask: ``flask._app_ctx_stack`` and ``flask._request_ctx_stack`` were deprecated in flask v2.2.2. These attributes were removed from flask integration.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/remove-deprecated-flask-methods-8d7a4a3d32c3c69d.yaml:4]:
```suggestion
    flask: remove deprecated usages of ``flask._app_ctx_stack`` and ``flask._request_ctx_stack``.
```

File: releasenotes/notes/remove-deprecated-flask-methods-8d7a4a3d32c3c69d.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    flask: add support for flask v2.3. Remove the following deprecated attributes: ``flask._app_ctx_stack`` and ``flask._request_ctx_stack``.
```
Comment (mabdinur) [inline releasenotes/notes/remove-deprecated-flask-methods-8d7a4a3d32c3c69d.yaml:4]:
```suggestion
    flask: add support for flask v2.3. Remove deprecated usages of ``flask._app_ctx_stack`` and ``flask._request_ctx_stack``.
```

**Review by brettlangdon — APPROVED** [review]

**Comment by mabdinur** [comment]
@Mergifyio backport 1.6 1.5

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**